src/System.Security.Cryptography.Algorithms/tests/HKDFTests.cs

```diff
@@ -607,5 +393,162 @@ public struct Rfc5869TestCase
+            {
+                byte[] prk = new byte[20];
+                AssertExtensions.Throws<ArgumentOutOfRangeException>(
+                    "outputLength",
+                    () => HKDF.Expand(HashAlgorithmName.SHA1, prk, 8421505, Array.Empty<byte>()));
+            }
+
+            [Fact]
+            public void Rfc5869DeriveKeyNullIkm()
+            {
+                AssertExtensions.Throws<ArgumentNullException>(
+                    "ikm",
+                    () => HKDF.DeriveKey(HashAlgorithmName.SHA1, null, 20, Array.Empty<byte>(), Array.Empty<byte>()));
+            }
+
+            [Fact]
+            public void Rfc5869DeriveKeyOkmMaxSizePlusOne()
+            {
+                byte[] ikm = new byte[20];
+                AssertExtensions.Throws<ArgumentOutOfRangeException>(
+                    "outputLength",
+                    () => HKDF.DeriveKey(HashAlgorithmName.SHA1, ikm, 20 * 255 + 1, Array.Empty<byte>(), Array.Empty<byte>()));
+            }
+
+            [Fact]
+            public void Rfc5869DeriveKeyOkmPotentiallyOverflowingValue()
+            {
+                byte[] ikm = new byte[20];
+                AssertExtensions.Throws<ArgumentOutOfRangeException>(
+                    "outputLength",
+                    () => HKDF.DeriveKey(HashAlgorithmName.SHA1, ikm, 8421505, Array.Empty<byte>(), Array.Empty<byte>()));
+            }
+        }
+
+        public class HkdfSpanTests : HKDFTests
+        {
+            protected override byte[] Extract(HashAlgorithmName hash, int prkLength, byte[] ikm, byte[] salt)
+            {
+                byte[] prk = new byte[prkLength];
+                Assert.Equal(prkLength, HKDF.Extract(hash, ikm, salt, prk));
+                return prk;
+            }
+
+            protected override byte[] Expand(HashAlgorithmName hash, byte[] prk, int outputLength, byte[] info)
+            {
+                byte[] output = new byte[outputLength];
+                HKDF.Expand(hash, prk, output, info);
+                return output;
+            }
+
+            protected override byte[] DeriveKey(HashAlgorithmName hash, byte[] ikm, int outputLength, byte[] salt, byte[] info)
+            {
+                byte[] output = new byte[outputLength];
+                HKDF.DeriveKey(hash, ikm, output, salt, info);
+                return output;
+            }
+
+            [Fact]
+            public void Rfc5869ExtractPrkTooLong()
+            {
+                byte[] prk = new byte[24];
+
+                for (int i = 0; i < 4; i++)
+                {
+                    prk[20 + i] = (byte)(i + 5);
+                }
+
+                byte[] ikm = new byte[20];
+                byte[] salt = new byte[20];
+                Assert.Equal(20, HKDF.Extract(HashAlgorithmName.SHA1, ikm, salt, prk));
+                Assert.Equal("A3CBF4A40F51A53E046F07397E52DF9286AE93A2".HexToByteArray(), prk.Take(20).ToArray());
```

Comment:
I think this is the only HexToByteArray vs ByteArrayToHex that got missed.  It'd be nice to fix it, but isn't a deal-breaker.